### ISSUE TITLE
Support non-ASCII data in header fields (RFC 2047)

### ISSUE THREAD

**Issue by robinkrahl** [aeshirey/EmlParser#7]
Header fields may contain non-ASCII data as defined in [RFC 2047](https://tools.ietf.org/html/rfc2047), for example `=?UTF-8?Q?Sch=C3=B6ne_Gr=C3=BC=C3=9Fe?=` is `Schöne Grüße`.  Would you accept a PR that adds support for RFC 2047?

**Comment by aeshirey** [comment]
> Would you accept a PR that adds support for RFC 2047?

Absolutely

**Comment by Esc4iCEscEsc** [comment]
Would love to see this supported as well!

**Comment by simnalamburt** [comment]
See also: https://github.com/soywod/rfc2047-decoder

**Comment by aeshirey** [comment]
Thanks for pointing me to that crate, @simnalamburt. I created the [`rfc2047` branch](https://github.com/aeshirey/EmlParser/tree/rfc2047) which [adds in RFC2047 parsing for non-email header fields](https://github.com/aeshirey/EmlParser/blob/cb2fe3b63aab1cb16b5f2c0a46927932fa3e6547/src/parser.rs#L142). I'm not sure whether email addresses can also be encoded in this way?

Existing tests still work, but I have no example yet to test this out. I can fabricate one, but if any of you have an existing example test case that would work, I'd happily use that.

While I'm at it, I also fixed a number of clippy lints and bumped the Rust edition to 2021.
